Marlin/clean_nozzle.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Marlin 3D Printer Firmware
+ * Copyright (C) 2016 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (C) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef __CLEAN_NOZZLE_H__
+#define __CLEAN_NOZZLE_H__
+
+#include "point_t.h"
+
+extern void do_blocking_move_to(float x, float y, float z);
```

Comment:
I agree we should avoid optimizing for speed where it matters little. But we should optimize for size at every opportunity. In some cases optimizing for size means replacing inline code with function calls, sacrificing a little performance, and I agree that is fine where code performance hardly matters, such as in the LCD and leveling routines. Easier-to-read code in homing and leveling will be better, even if it's slower.

I also agree, movement should be available all over the place, wherever we might need to utilize it in other classes. We can probably break out a lot of the general-purpose functions into a separate compilation unit from `Marlin_main`. In fact, I'm sure we will end up splitting it into a few different files based on types of functions - temperature, movement, etc.